src/plugins/mf_priority.cpp

```diff
@@ -950,6 +954,49 @@ static int validate_cb (flux_plugin_t *p,
                                            "user has max active jobs");
     }
 
+    if (jobspec == NULL) {
+        flux_jobtap_reject_job (p,
+                                args,
+                                "job.validate: failed to unpack jobspec");
+    } else {
+        // if a queue was not unpacked, just set it to ""
+        queue_str = queue ? queue : "";
+        // count resources requested for the job
+        if (job.count_resources (jobspec) < 0) {
+            return flux_jobtap_reject_job (p,
+                                           args,
+                                           "job.validate: unable to count " \
+                                           "resources for job");
+        }
+        if (job.ncores > 0 && job.nnodes == 0)
+            // the job specified cores but no nodes, so we need to set
+            // nnodes == 1 here
+            job.nnodes = 1;
+        // look up queue in queues map to see if it has a defined
+        // max_nodes_per_association limit
+        auto it = queues.find(queue_str);
+        if (it != queues.end ()) {
+            if (job.nnodes > queues[queue_str].max_nodes_per_assoc) {
+            // the job size is greater than the max nodes per-association limit
+            // configured for this queue; reject the job
+            return flux_jobtap_reject_job (p,
+                                           args,
+                                           "job.validate: job size is greater "
+                                           "than max resources limit "
+                                           "configured for queue");
+            }
+        }
+        if ((job.nnodes > a->max_nodes) || (job.ncores > a->max_cores)) {
+            // the job size is greater than the max resources limits (max nodes
+            // OR max cores) configured for this association; reject the job
+            return flux_jobtap_reject_job (p,
+                                           args,
+                                           "job.validate: job size is greater "
+                                           "than max resources limits "
+                                           "configured for association");
+        }
```

Comment:
similar to above, would it make sense to include the detected job size and the limit (or at least just the limit). Also, I suggest dropping the `job.validate: ` prefix.